compiler/rustc_mir_build/src/build/block.rs

```diff
@@ -109,13 +109,102 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                         )
                     );
                 }
+                StmtKind::Let {
+                    remainder_scope,
+                    init_scope,
+                    pattern,
+                    initializer: Some(initializer),
+                    lint_level,
+                    else_block: Some(else_block),
+                } => {
```

Comment:
It honestly would be helpful to either have a test with mir output that covers this, or an example with the expected mir output in comments here.

Reply:
Let me try both. :smile:

Reply:
Speeking too soon. It seems not possible to dump MIR in `ui` tests yet. I have added explanatory comments here instead.

Reply:
There is a list of MIR based tests in `src/test/mir-opt`. Despite the name of the directory, not all are about optimization. You can check out `src/test/mir-opt/storage_ranges.rs` for an example that just checks the generated MIR. It is also explained [here](https://rustc-dev-guide.rust-lang.org/tests/compiletest.html#mir-opt-tests).